### PR TITLE
Tidy up test output and prevent new noise

### PR DESCRIPTION
**Pull request by drew-walker**
I think this is a terrible practice, but I also think it might be a good way to stem the growing number of warnings and errors in tests. This change will mean new categories of errors or warnings will fail the test, and then the patterns from setupTests.js can be addressed as debt items at a later date. Here's how the test output looks with these changes:

Screen Shot 2020-05-22 at 9 18 58 PM

### CODE REVIEW

File: packages/@react-spectrum/link/test/Link.test.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ describe('Link', function () {
   let onPressSpy = jest.fn();
 
   afterEach(() => {
+    cleanup();
```
Comment (snowystinger) [inline packages/@react-spectrum/link/test/Link.test.js:30]: why was this added? it should be enforced now

Comment (drew-walker) [reply packages/@react-spectrum/link/test/Link.test.js:30]: I believe as of now that every call to render appends to document.body, which means when we call `let link = getByRole('link');` we're actually selecting 2 links from different tests. That ends up throwing a warning, and could potential result in false positives in tests.

File: setupTests.js [hunk 2]
```diff
@@ -12,3 +12,56 @@
...
+  'Received NaN for the `%s` attribute',
+  'Can\'t perform a React state update on an unmounted component',
+  '%s contains an input of type %s with both value and defaultValue props',
+  '<%s /> is using incorrect casing',
+  'The tag <%s> is unrecognized in this browser',
+  'The above error occurred in the <TestHook> component'
+];
+
+const WARNING_PATTERNS_WE_SHOULD_FIX_BUT_ALLOW = [
+  '`waitForDomChange` has been deprecated.',
```
Comment (drew-walker) [inline setupTests.js:30]: This might be a relatively straightforward warning to get through and fix.

File: packages/@react-spectrum/link/test/Link.test.js (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/link/test/Link.test.js:30]: where are we rendering two links?
if you're referring to it.each, i believe every iteration of each is considered it's own test and should get cleanup automatically applied

Comment (drew-walker) [reply packages/@react-spectrum/link/test/Link.test.js:30]: I'm very confused now. I can't reproduce this anymore. I was originally seeing a warning when calling `getByRole`. When debugging, I could see the 2 links, the one from the 2nd-last test, and from the last test. That's what led me down calling `cleanup()` in `afterEach()`. According to the docs it should definitely be calling cleanup automatically in the afterEach block, so I'll remove this addition.

File: packages/@react-spectrum/link/test/Link.test.js [hunk 3]
```diff
@@ -10,7 +10,7 @@
  * governing permissions and limitations under the License.
  */
 
-import {fireEvent, render} from '@testing-library/react';
+import {cleanup, fireEvent, render} from '@testing-library/react';
```
Comment (snowystinger) [inline packages/@react-spectrum/link/test/Link.test.js:13]: unused import now :)

**Comment by snowystinger** [comment]
here's where I'm at for fixing these. i do like this, because it made it easy to break down and focus on classes of errors in small PR's (so far, I've left the act ones alone for the moment, haha)
```
const ERROR_PATTERNS_WE_SHOULD_FIX_BUT_ALLOW = [
  'An update to %s inside a test was not wrapped in act',

  'Unknown event handler property `%s`',

  // https://github.com/adobe-private/react-spectrum-v3/pull/510
  'Not implemented: navigation',

  // https://github.com/adobe-private/react-spectrum-v3/pull/509
  'React does not recognize the `%s` prop on a DOM element',

  // https://github.com/adobe-private/react-spectrum-v3/pull/508
  'Received NaN for the `%s` attribute',

  'Can\'t perform a React state update on an unmounted component',

  '%s contains an input of type %s with both value and defaultValue props',

  // https://github.com/adobe-private/react-spectrum-v3/pull/511
  '<%s /> is using incorrect casing',

  'The tag <%s> is unrecognized in this browser',

  // https://github.com/adobe-private/react-spectrum-v3/pull/512
  'The above error occurred in the <TestHook> component'
];

const WARNING_PATTERNS_WE_SHOULD_FIX_BUT_ALLOW = [
  '`waitForDomChange` has been deprecated.',

  // AlertDialog (and TableCrud which uses AlertDialog) and IllustratedMessage should be fixed by https://github.com/adobe-private/react-spectrum-v3/pull/501
  'The style prop is unsafe and is unsupported in React Spectrum v3.',

  // Test which mistakenly used classname should be fixed by https://github.com/adobe-private/react-spectrum-v3/pull/501 as well
  'The className prop is unsafe and is unsupported in React Spectrum v3.',

  // https://github.com/adobe-private/react-spectrum-v3/pull/507
  'If you do not provide a visible label, you must specify an aria-label or aria-labelledby attribute for accessibility'
];
```

File: packages/@react-spectrum/link/test/Link.test.js (hunk 3 above)
Comment (drew-walker) [reply packages/@react-spectrum/link/test/Link.test.js]: 🤦

File: packages/@react-spectrum/listbox/test/ListBox.test.js [hunk 4]
```diff
@@ -530,7 +530,7 @@ describe('ListBox', function () {
     let tree = render(
       <Provider theme={theme}>
         <ListBox>
-          <Item>
+          <Item textValue="Label">
```
Comment (snowystinger) [inline packages/@react-spectrum/listbox/test/ListBox.test.js:533]: this should probably be aria-label instead of textValue

File: setupTests.js (hunk 2 above)
Comment (snowystinger) [reply setupTests.js:30]: probably, do you want to do it here? (i'm fine if you don't)

File: setupTests.js [hunk 5]
```diff
@@ -12,3 +12,56 @@
...
+  'Unknown event handler property `%s`',
+  'Not implemented: navigation',
+  'React does not recognize the `%s` prop on a DOM element',
+  'Received NaN for the `%s` attribute',
+  'Can\'t perform a React state update on an unmounted component',
+  '%s contains an input of type %s with both value and defaultValue props',
+  '<%s /> is using incorrect casing',
+  'The tag <%s> is unrecognized in this browser',
+  'The above error occurred in the <TestHook> component'
+];
```
Comment (LFDanLu) [inline setupTests.js:27]: Since @snowystinger has addressed a bunch of these in some follow up pulls, should we strip them out of the array or do we want to keep this as is? Fine with it either way

Comment (drew-walker) [reply setupTests.js:27]: I'm happy to update this PR with the minimal set once the PRs from @snowystinger are merged.

File: setupTests.js (hunk 2 above)
Comment (drew-walker) [reply setupTests.js:30]: I'll take another look and if I can solve it in an hour or two I'll include the fixes in this PR.

File: setupTests.js (hunk 5 above)
Comment (snowystinger) [reply setupTests.js:27]: I wouldn't worry about it in this PR, we can follow up with stripping them out

File: packages/@react-spectrum/listbox/test/ListBox.test.js (hunk 4 above)
Comment (snowystinger) [reply packages/@react-spectrum/listbox/test/ListBox.test.js:533]: ignore this, i got more clarification on it, this is right

**Comment by snowystinger** [comment]
ugh... that push did not work right, i'll just remake the branch
